### PR TITLE
Added cancellation of requests using CancellationToken

### PR DESCRIPTION
**Pull request by aKzenT**
Please see related issue #21

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hello @aKzenT,<br /><br/>Thank you for the PR!<br />                                              <br />We will be reviewing this PR soon, in the mean                                              time, if you have not, please be sure to                                              <a href="https://github.com/sendgrid/csharp-http-client/blob/master/CONTRIBUTING.md#cla">                                             sign our CLA</a> so that we can merge your changes. Thanks again!                                             <br /><br />Team DX

**Comment by thinkingserious** [comment]
Hello @aKzenT,

Do you mind sending over a [signed CLA](https://github.com/sendgrid/csharp-http-client/blob/master/CONTRIBUTING.md#cla) so I can review and merge?

Thanks!

**Comment by aKzenT** [comment]
Already did that a few weeks ago. Please Check!

**Comment by thinkingserious** [comment]
Thanks @aKzenT,

I do not see it however, can you please let me know from what email you sent it from? (Please reply to dx@sendgrid.com)

**Comment by thinkingserious** [comment]
Hi @aKzenT,

I have not received the CLA or follow up email from you. Could you please resend or let me know what email address you used to send it? Thanks!

**Comment by aKzenT** [comment]
I send it again, now from another email address and to your own email address just in case.

**Comment by thinkingserious** [comment]
@aKzenT,

We received it, thanks!

As for v9, it's available now:
https://github.com/sendgrid/sendgrid-csharp
https://www.nuget.org/packages/SendGrid

With Best Regards,

Elmer

File: CSharpHTTPClient/Client.cs [hunk 1]
```diff
@@ -278,9 +280,13 @@ public override bool TryInvokeMember(InvokeMemberBinder binder, object[] args, o
...
                         AddRequestHeader((Dictionary<string, string>)obj);
                     }
+                    else if (name == "cancellationToken")
+                    {
+                        cancellationToken = (CancellationToken)obj;
+                    }
                     i++;
                 }
-                result = RequestAsync(binder.Name.ToUpper(), requestBody: requestBody, queryParams: queryParams).ConfigureAwait(false);
+                result = RequestAsync(binder.Name.ToUpper(), cancellationToken, requestBody: requestBody, queryParams: queryParams).ConfigureAwait(false);
```
Comment (thinkingserious) [inline CSharpHTTPClient/Client.cs]: Please move the `cancellationToken` parameter to the end for consistency with our v9 SDK.

File: CSharpHTTPClient/Client.cs [hunk 2]
```diff
@@ -296,21 +302,23 @@ public override bool TryInvokeMember(InvokeMemberBinder binder, object[] args, o
...
         /// <summary>
         ///     Prepare for async call to the API server
         /// </summary>
         /// <param name="method">HTTP verb</param>
+        /// <param name="cancellationToken">A token that allows cancellation of the http request</param>
         /// <param name="requestBody">JSON formatted string</param>
         /// <param name="queryParams">JSON formatted queary paramaters</param>
         /// <returns>Response object</returns>
-        private async Task<Response> RequestAsync(string method, string requestBody = null, string queryParams = null)
+        private async Task<Response> RequestAsync(string method, CancellationToken cancellationToken, string requestBody = null, string queryParams = null)
```
Comment (thinkingserious) [inline CSharpHTTPClient/Client.cs]: Please move the `cancellationToken` parameter to the end for consistency with our v9 SDK.

**Review by thinkingserious — CHANGES_REQUESTED** [review]

**Comment by thinkingserious** [comment]
Hi @aKzenT,

To merge this, could you please:

1. Resolve the conflicts
2. Give the cancellation token a default value so we don't have to make a breaking change

Thanks!

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/csharp-http-client?pullRequest=22) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/csharp-http-client?pullRequest=22) before we can accept your contribution.<br/><hr/>**Thomas Krause** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/csharp-http-client?pullRequest=22) it.</sub>

**Review by thinkingserious — APPROVED** [review]

**Comment by thinkingserious** [comment]
Hello @aKzenT,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by aKzenT** [comment]
Done, Thanks!

**Pull request merged**